lib/Conversion/TritonGPUToLLVM/Patterns.h

```diff
@@ -0,0 +1,94 @@
+#ifndef TRITON_CONVERSION_TRITONGPU_TO_LLVM_PATTERNS_H
```

Comment:
Up to you, but personally I don't love having one header file that corresponds to multiple .cc files.  Makes things confusing to edit, because the mapping between files is not obvious.

If you want one header (I assume this is for simplicity?) I'm not sure what is the goal behind having multiple cc files -- is it compile time?

At least maybe we could give the cc files names that more obviously relate to the header file?  Like, `Patterns.h` and `PatternLoadStoreOpToLLVM.cc` or something?

Reply:
My motivation was mostly 1. following the convention of how things are in MLIR in general, 2. avoid having many headers with just one populate declaration. Having multiple .cc is needed to keep the code partitioned as having one file would be gigantic.

It looks common in LLVM for passes and other such thing where the header is small but the implementation is big.

Eventually I think we could have multiple headers to separate what is target agnostic and what is target specific. 

How about naming the header `PatternTritonGPUToLLVM.h`? What do you think?

Reply:
> How about naming the header PatternTritonGPUToLLVM.h? What do you think?

I think if when I typed `ls` in the directory and the .h and .cc files all showed up next to each other (because they share a prefix) then that wouldn't be too confusing to me.

Reply:
I guess here the .cpp file don't have a common prefix, although in this folder most file are patterns so I'm not sure what to do.
I wonder if others have a preference?

Reply:
Sure, another way would be to have a `patterns` folder, then the filenames don't matter.

Reply:
Anyway, I don't block the review on it, this is just my preference.

Reply:
Yeah having a folder for patterns would make sense. Ok I'll leave this open until end of day to see if others want to weight in if not I would probably just rename the file `PatternTritonGPUOpToLLVM.h` and go with this, there will probably be more refactoring as we try to unify those patterns with other backends.

Reply:
I'm OK with the change. `PatternTritonGPUOpToLLVM` seems better to me